### ISSUE TITLE
RangeError while checking dependencies

### ISSUE THREAD

**Issue by yshrsmz** [zbigg/copy-production-deps#4]
while testing the package, I ran into the following error

```
yshrsmz: ~/repos/test% npx copy-production-deps packages/media dist -v
npx: Installed 41 packages in 2.034 seconds.
copy-production-deps: collecting production packages for packages/media
copy-production-deps: failed: RangeError: Maximum call stack size exceeded RangeError: Maximum call stack size exceeded
    at internal/fs/utils.js:547:42
    at hidden (internal/errors.js:290:14)
    at Object.openSync (fs.js:450:10)
    at Object.readFileSync (fs.js:360:35)
    at loadJsonSync (/Users/a12897/.npm/_npx/73927/lib/node_modules/copy-production-deps/lib/copy-production-deps.js:52:36)
    at lookForDependenciesInWorkspace (/Users/yshrsmz/.npm/_npx/73927/lib/node_modules/copy-production-deps/lib/copy-production-deps.js:106:46)
    at processPackage (/Users/yshrsmz/.npm/_npx/73927/lib/node_modules/copy-production-deps/lib/copy-production-deps.js:151:5)
    at lookForDependenciesInWorkspace (/Users/yshrsmz/.npm/_npx/73927/lib/node_modules/copy-production-deps/lib/copy-production-deps.js:139:9)
    at processPackage (/Users/yshrsmz/.npm/_npx/73927/lib/node_modules/copy-production-deps/lib/copy-production-deps.js:151:5)
    at lookForDependenciesInWorkspace (/Users/yshrsmz/.npm/_npx/73927/lib/node_modules/copy-production-deps/lib/copy-production-deps.js:139:9)
```

**Comment by zbigg** [comment]
@yshrsmz Looks like you have cycle in folder structure of packages and code doesn't handle that :/. While writing this i wasn't never noticed such folder structure.

Unfortunately, there is no builtin debug facility to tell which package/folder triggers this behavior.

Is there any chance for link where i can reproduce this issue ?

Hint - You may use `strace -ff` or explicitly add `console.log in `loadJsonSync` in `node_modules/copy-production/deps/lib/copy-productions-deps.js` to debug it and pinpoint the package that has cycling dependency to itself.

**Comment by zbigg** [comment]
> While writing this i never noticed such folder structure ... and i'm wondering how it's even possible.

Of course, code need to be protected against cycle in folder structure and i this needs fix.

**Comment by yshrsmz** [comment]
Hi, thanks for your quick response!
Unfortunately, it's a private project so I can not share it with you but I'm going to try to create a repro. 

In the meantime, I added `console.log` in `loadJsonSync` as you suggested and got huge 7MB output.
[out.txt.zip](https://github.com/zbigg/copy-production-deps/files/4959884/out.txt.zip)

**Comment by zbigg** [comment]
@yshrsmz Thanks for output. I think i've identified package with cycle in dependency tree. Namely: `string.prototype.trimstart` & `es-abstract`.

See this: http://npm.broofa.com/?q=string.prototype.trimstart

I've got repro with empty repo containing:
```
"dependencies": {
    "string.prototype.trimstart": "^1.0.1"
  }
```

With this one, fix should be quite easy.

**Comment by zbigg** [comment]
@yshrsmz Fix for this particular issue released in [0.1.6](https://github.com/zbigg/copy-production-deps/releases/tag/v0.1.6) just published.

However there is still a bug that prevents working with your repro repo: https://github.com/yshrsmz/nuxt-serverless-test

See #6

**Comment by yshrsmz** [comment]
Thanks!
#6 seems to happen in my previously-mentioned repo too, as it also uses Nuxt.js.
Hope that can be fixed too.

**Comment by zbigg** [comment]
@yshrsmz #6 Is fixed in [0.1.7](https://github.com/zbigg/copy-production-deps/releases/tag/v0.1.6) ... also already published on npm. If you still have patience, please give it a try.

**Comment by yshrsmz** [comment]
@zbigg Hi, I can confirm that the command now finishes without throwing error!
I can not fully check the output for now but will do it next Monday and get back to you if I find anything strange.

Thanks for the useful package!
